Add various dev filters to Sharing_Source::get_link()

clifgriffin opened the pull request:
Howdy,

Recently had to make some modifications to output of the sharing buttons for a client, and found that the provided filters were a bit weak. 

I've added a few that I feel provide as much versatility as possible without doing anything obnoxious. 

The only unorthodox thing I'm doing here is passing all of the original arguments to each filter using `func_get_args()`.  This is so that developers always have the full context.  (I've verified this function is safe with PHP 5.2+)

However, a more complete implementation might need to include full and immediate filtering of all arguments, pre-output or manipulation. Something like:
`extract ( apply_filters('sharing_display_args', compact('url', 'text', 'title', 'query', 'id') ) );`

Which, I didn't include because I'm fairly certain `extract()` is not kosher even in this application :)

Please let me know if you'd prefer any adjustments, or if you have any thoughts on the above.  I can add via a second pull request if desired. 

Clif

samhotchkiss commented:
Hey Clif-- this looks good, the only change I'd recommend on first pass is namespacing the filters.  Since we have to keep the existing filter name intact for backwards compatibility, I would leave it as is, then add your 6 new filters after it, prefixed by `jetpack_`  (giving us two filters on url)

clifgriffin commented:
@samhotchkiss I namespaced everything.  I also added a filter for jetpack_sharing_display_link, but I added it in addition to the old one for backwards compatibility and commented it thusly. 

Let me know if you need any other tweaks.
